src/backend/distributed/executor/intermediate_results.c

```diff
@@ -191,6 +191,19 @@ CreateRemoteFileDestReceiver(char *resultId, EState *executorState,
 	resultDest->memoryContext = CurrentMemoryContext;
 	resultDest->writeLocalFile = writeLocalFile;
 
+	/*
+	 * Make sure that this transaction has a distributed transaction ID.
+	 *
+	 * Intermediate results will be stored in a directory that is derived from
+	 * the distributed transaction ID across all workers and on the coordinator
+	 * itself. Even if we only store results locally, we still want to assign
+	 * a transaction ID in case we later store results on workers.
+	 *
+	 * When we start using broadcast_intermediate_result from workers, we
+	 * need to make sure that we don't override the transaction ID here.
+	 */
+	BeginOrContinueCoordinatedTransaction();
```

Comment:
I suppose it would be a strange side-effect. Added a comment that the caller should do this.

I added it to `create_intermediate_result` as well, though it's kind of unnecessary there since it will just use local transaction scope.